src/main.zig

```diff
@@ -3020,6 +3020,32 @@ fn buildOutputType(
         break :l global_cache_directory;
     };
 
+    for (c_source_files.items) |*src| {
+        if (!mem.eql(u8, src.src_path, "-")) continue;
+
+        const ext = src.ext orelse
+            fatal("-E or -x is required when reading from a non-regular file", .{});
+
+        // "-" is stdin. Dump it to a real file.
+        const sep = fs.path.sep_str;
+        const sub_path = try std.fmt.allocPrint(arena, "tmp" ++ sep ++ "{x}-stdin{s}", .{
+            std.crypto.random.int(u64), ext.canonicalName(target_info.target),
+        });
+        try local_cache_directory.handle.makePath("tmp");
+        // Note that in one of the happy paths, execve() is used to switch
+        // to clang in which case any cleanup logic that exists for this
+        // temporary file will not run and this temp file will be leaked.
+        // Oh well. It's a minor punishment for using `-x c` which nobody
+        // should be doing. Therefore, we make no effort to clean up. Using
+        // `-` for stdin as a source file always leaks a temp file.
```

Comment:
we could minimize the damage by:
- hashing the file contents (while writing to the file, so no additional allocations)
- renaming the filename to the hash.

That way, we avoid a small chance of hash collision (I know) and multiple invocations of `echo | zig cc <...>` produce only one garbage-file.

Thoughts?

Reply:
OK, sounds reasonable

Reply:
https://github.com/ziglang/zig/pull/15770 and https://github.com/ziglang/zig/pull/15772